packages/components/src/components/floor-selector/floor-selector.tsx

```diff
@@ -0,0 +1,237 @@
+            const listPaddingTop: number = parseInt(window.getComputedStyle(this.floorListElement).getPropertyValue('padding-top'));
+            const seletedElementDistanceFromTop = this.currentFloorElement.offsetTop - this.floorListElement.offsetTop;
+            const distance: number = -(seletedElementDistanceFromTop - listScrollTop - listPaddingTop);
+            this.animateTranslateY(this.currentFloorElement, distance, 250);
+
+            // If the floor selector has been scrolled, all floors need to be pushed down.
+            if (this.floorListElement.scrollTop > 0) {
+                floorElements.forEach(floorElement => {
+                    if (floorElement.getAttribute('data-floor') !== this.currentFloor) {
+                        // Floors placed before the first floor need to be pushed further down.
+                        if (elementsBeforeSelected.indexOf(floorElement) > -1) {
+                            this.animateTranslateY(floorElement as HTMLElement, (floorElement as HTMLElement).offsetTop + this.floorListElement.scrollTop, 250);
+                        } else {
+                            this.animateTranslateY(floorElement as HTMLElement, this.floorListElement.scrollTop, 250);
+                        }
+                    }
+                });
+            } else {
+                // If the floor selector has NOT been scrolled, only the elements above the selected floor need to be pushed down.
+                // The selected floor will take the position of the first floor in the list.
+                elementsBeforeSelected.forEach(floorElement => {
+                    this.animateTranslateY(floorElement, floorElement.offsetTop, 250);
+                });
+            }
+        } else {
+            // If the floor selector is closed, all elements need to be animated back to their original position.
+            this.animateTranslateY(this.currentFloorElement, 0, 250);
+            floorElements.forEach(floorElement => {
+                if (floorElement.getAttribute('data-floor') !== this.currentFloor) {
+                    this.animateTranslateY(floorElement as HTMLElement, 0, 250);
+                }
+            });
+        }
+    }
+
+    /**
+     * Toggles the floor selector open-close.
+     */
+    private onToggleFloorSelector(): void {
+        this.floorSelectorClosed = !this.floorSelectorClosed;
+        this.animateFloorSelector();
+
+        if (!this.floorSelectorClosed) {
+            this.scrollToSelectedFloor();
+        }
+
+        this.onScrollStyle();
+    }
+
+    /**
+     * Adding/removing classes to the floorlist to fade-in/out the top/bottom part of the container..
+     */
+    private onScrollStyle(): void {
+        const maxScrollTop = this.floorListElement.scrollHeight - this.floorListElement.clientHeight;
+
+        if (this.floorSelectorClosed) {
+            this.floorListElement.classList.remove('mi-floor-selector__list--fade-top');
```

Comment:
Can we add and remove classes in the JSX instead of using classList? I imagine the JSX could react to some variable telling what class names to render.

Reply:
@ammapspeople this has also been resolved by introducing 2 properties: `fadeFloorSelectorOnTop` and `fadeFloorSelectorOnBottom` to control the adding/removing of classes